### ISSUE TITLE
[no-unused-expressions] Optional Chaining returning false positive

### ISSUE THREAD

**Issue by vmuresanu** [typescript-eslint/typescript-eslint#2086]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-unused-expressions": [true]
  }
}
```

Consider following scenarios: 

A:
```TS
---
function declaration
x(a) {
    a?.b();
  }
---
function call
x ( { b: () => '1' });
```

B:
```TS
---
function declaration
x(a) {
    a.b();
  }
---
function call
x ( { b: () => '1' });
```

C:
```TS
---
function declaration
x(a) {
   if(a) {
      a.b();
   } 
  }
---
function call
x ( { b: () => '1' });
```

**Expected Result**
Case A, B and C - passes

**Actual Result**
Case A returns ```error  Expected an assignment or function call and instead saw an expression  no-unused-expressions```
Case B - passes 
Case C - passes

**Additional Info**
Case C is a workaround for case A that I am currently using, as in Case B parameter 'a' could be undefined thus I need to make a check. Option C was the 'natural' way of handling this scenarios, but now with Typescript's 3.7 version Optional Chaining operator it is syntactic sugar to bypass these if checks. 

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.34.0` |
| `@typescript-eslint/parser`        | `2.34.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `13.12.0` |
| `npm`                              | `6.14.4` |

**Comment by bradzacher** [comment]
I'm not sure I understand what's wrong.
Why should case A error? Or did you get expected and actual the wrong way around.

**Comment by vmuresanu** [comment]
@bradzacher sorry, I got expected and actual the wrong way. I've edited it. Thank you for your observation

**Comment by bradzacher** [comment]
I cannot repro this - we have passing test for this against master.
https://github.com/typescript-eslint/typescript-eslint/blob/014341bb23261f609fc2a6fe7fece191466a084a/packages/eslint-plugin/tests/rules/no-unused-expressions.test.ts#L42-L47

Could you please provide a repro repo?

**Comment by vmuresanu** [comment]
@bradzacher yes looking into it now.

**Comment by vmuresanu** [comment]
@bradzacher ok I couldn't use plunkr or jsFiddle as eslint requires npm scripts to run. So here is my repo - https://github.com/vmuresanu/launch-media-app. Here is my PR in which I added eslint and a method which reproduces the issue - https://github.com/vmuresanu/launch-media-app/pull/2 so that you can see the changes I added (btw I am using eslint 7.1.0 and @typescript-eslint/eslint-plugin" and parser: "3.0.0". 
The repro code is in ```host.component.ts``` 

Steps: 
1. In package json run ```npm run lint``` 
You will see - ```error    Expected an assignment or function call and instead saw an expression  no-unused-expressions```
2. Remove the optional chaining operator from line 29. Rerun lint
3. No error occurs.

**Comment by bradzacher** [comment]
https://github.com/vmuresanu/launch-media-app/blob/master/.eslintrc.js#L20

You're using the wrong rule.

https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md#i-am-using-a-rule-from-eslint-core-and-it-doesnt-work-correctly-with-typescript-code

**Comment by vmuresanu** [comment]
Works thank you! :) @bradzacher
